### DISCUSSION TITLE
What's the output of wtextimage?

### DISCUSSION THREAD

**Discussion by wuqso** [iraf-community/iraf#401] [Q&A]
I used wtextimage to convert an IRAF image (1-D spectrum) into a text file and I got 5 columns of numbers as follows,

...
  1.713505E-16  2.057833E-16  2.210677E-16  2.126595E-16  1.783040E-16
  1.828803E-16  2.377377E-16  2.025416E-16  2.286014E-16  1.988441E-16
  1.746359E-16  2.029601E-16  1.971067E-16  1.760395E-16  1.854584E-16
  1.573641E-16  1.902356E-16  2.273310E-16  1.731561E-16  1.787992E-16
  1.891723E-16  1.735168E-16  1.931826E-16  1.961806E-16  1.954478E-16
  2.234496E-16  1.922625E-16  1.898011E-16  2.206757E-16  2.060543E-16
...

What are these numbers? If I don't make a mistake, the wtextimage should produce a text file with columns as wavelength, flux, and flux error, ...

**Comment by olebole** [comment]
The numbers are just the pixel values of your file, as documented in the [help file of wtextimage](https://iraf.readthedocs.io/en/latest/tasks/dataio/wtextimage.html):
> IRAF images are converted to text files with procedure wtextimage. The text file written consists of an optional header optionally followed by the pixel values. The pixels are output in FITS order, that is, the leftmost subscript varies most rapidly. The image header is written in the "keyword = value / comment" format of FITS.

**Comment by wuqso** [comment]
Then what task can change a fits image of a 1-D spectrum into a text file, so I can plot the spectrum with other softwares/apps?

**Comment by wuqso** [comment]
I got it. the wspectext in onedspec can do this. But the flux didn't have an error...

**Reply by olebole** [reply]
If it is a 1D image, it probably does not come with the errors. BTW, why do you need to do this with IRAF? Reading a FITS 1D image and writing it to text is simple (and more future-proof) with Astropy.

I would not start solving problems with IRAF; IMO it is only useful for legacy, or when there is no other way is available.
